src/AppInstallerRepositoryCore/Rest/RestSource.cpp

```diff
@@ -0,0 +1,262 @@
+                {
+                    for (const auto& versionInfo : m_package.versions)
+                    {
+                        if (CaseInsensitiveEquals(versionInfo.GetVersion().ToString(), versionKey.Version)
+                            && CaseInsensitiveEquals(versionInfo.GetChannel().ToString(), versionKey.Channel))
+                        {
+                            packageVersion = std::make_shared<PackageVersion>(source, m_package.packageInfo, versionInfo);
+                        }
+                    }
+                }
+                else if (versionKey.Version.empty() && versionKey.Channel.empty())
+                {
+                    return GetLatestAvailableVersion();
+                }
+                else if (versionKey.Version.empty())
+                {
+                    for (const auto& versionInfo : m_package.versions)
+                    {
+                        if (CaseInsensitiveEquals(versionInfo.GetChannel().ToString(), versionKey.Channel))
+                        {
+                            packageVersion = std::make_shared<PackageVersion>(source, m_package.packageInfo, versionInfo);
+                        }
+                    }
+                }
+                else if (versionKey.Channel.empty())
+                {
+                    for (const auto& versionInfo : m_package.versions)
+                    {
+                        if (CaseInsensitiveEquals(versionInfo.GetVersion().ToString(), versionKey.Version))
+                        {
+                            packageVersion = std::make_shared<PackageVersion>(source, m_package.packageInfo, versionInfo);
+                        }
+                    }
+                }
+
+                return packageVersion;
+            }
+
+            bool IsUpdateAvailable() const override
+            {
+                return false;
+            }
+        };
+    }
+
+    RestSource::RestSource(const SourceDetails& details, std::string identifier, RestClient&& restClient)
+        : m_details(details), m_identifier(std::move(identifier)), m_restClient(std::move(restClient))
+    {
+    }
+
+    const SourceDetails& RestSource::GetDetails() const
+    {
+        return m_details;
+    }
+
+    const RestClient& RestSource::GetRestClient() const
+    {
+        return m_restClient;
+    }
+
+    const std::string& RestSource::GetIdentifier() const
+    {
+        return m_identifier;
+    }
+
+    SearchResult RestSource::Search(const SearchRequest& request) const
+    {
+        // Note: Basic search functionality to fetch everything.
+        RestClient::SearchResult results = m_restClient.Search(request);
+        SearchResult searchResult;
+
+        std::shared_ptr<const RestSource> sharedThis = shared_from_this();
+        for (auto& result : results.Matches)
+        {
+            std::unique_ptr<IPackage> package = std::make_unique<AvailablePackage>(sharedThis, result);

```

Comment:
Should `std::move` the result into the package since it will no longer be needed.